sym_api_client_python/datafeed_event_service.py

```diff
@@ -5,358 +5,63 @@
-            'ROOMMEMBERDEMOTEDFROMOWNER' : self.demoted_to_owner,
-            'CONNECTIONACCEPTED' : self.connection_accepted_handler,
-            'CONNECTIONREQUESTED' : self.connection_requested_handler,
-            'SYMPHONYELEMENTSACTION' : self.elements_action_handler,
-            'SHAREDPOST': self.shared_post_handler,
-        }
+            self.datafeed_event_service = DataFeedEventServiceV1(sym_bot_client, log_events=log_events,
+                                                                 error_timeout_sec=error_timeout_sec, maximum_timeout_sec=maximum_timeout_sec)
 
     def start_datafeed(self):
-        log.info('DataFeedEventService/startDataFeed()')
-        self.datafeed_id = self.datafeed_client.create_datafeed()
-        self.read_datafeed()
-
-    def add_room_listener(self, room_listener):
-        self.room_listeners.append(room_listener)
-
-    def remove_room_listener(self, room_listener):
-        self.room_listeners.remove(room_listener)
-
-    def add_im_listener(self, im_listener):
-        self.im_listeners.append(im_listener)
-
-    def remove_im_listener(self, im_listener):
-        self.im_listeners.remove(im_listener)
-
-    def add_connection_listener(self, connection_listener):
-        self.connection_listeners.append(connection_listener)
-
-    def remove_connection_listener(self, connection_listener):
-        self.connection_listeners.remove(connection_listener)
-
-    def add_elements_listener(self, elements_listener):
-        self.elements_listeners.append(elements_listener)
-
-    def remove_elements_listener(self, elements_listener):
-        self.elements_listeners.remove(elements_listener)
-
-    def add_wall_post_listener(self, wall_post_listener):
-        self.wall_post_listeners.append(wall_post_listener)
-
-    def remove_wall_post_listener(self, wall_post_listener):
-        self.wall_post_listeners.remove(self, wall_post_listener)
-
-    def add_suppression_listener(self, suppression_listener):
-        self.suppression_listeners.append(suppression_listener)
-
-    def remove_suppression_listener(self, suppression_listener):
-        self.suppression_listeners.remove(suppression_listener)
+        """Start reading events from datafeed.
+        """
+        self.datafeed_event_service.start_datafeed()
 
     def activate_datafeed(self):
-        if self.stop:
-            self.stop = False
+        self.datafeed_event_service.activate_datafeed()
 
     def deactivate_datafeed(self):
-        if not self.stop:
-            self.stop = True
-
-    # read_datafeed function reads an array of events coming back from
-    # DataFeedClient checks to see that sender's email is not equal to Bot's
-    # email in config filemthis functionality helps bot avoid entering an
-    # infinite loop where it responds to its own messageSent
-    # after an event comes back over the dataFeed, the json object returned
-    # from read_datafeed() gets passed to handle_event()
-    def read_datafeed(self):
```

Comment:
I saw that you removed many methods in this class, maybe this methods is no more needed in this class but this will break the backward compatibility.

Reply:
True ! I will implement the proper functions in the datafeed_event_service then.